plugins/woocommerce/src/Blocks/AssetsController.php

```diff
@@ -44,6 +45,28 @@ protected function init() { // phpcs:ignore WooCommerce.Functions.InternalInject
 		add_action( 'wp_enqueue_scripts', array( $this, 'update_block_settings_dependencies' ), 100 );
 		add_action( 'admin_enqueue_scripts', array( $this, 'update_block_settings_dependencies' ), 100 );
 		add_filter( 'js_do_concat', array( $this, 'skip_boost_minification_for_cart_checkout' ), 10, 2 );
+
+		if ( Features::is_enabled( 'experimental-iapi-runtime' ) ) {
+			// Run after the WordPress iAPI runtime has been registered by setting a lower priority.
+			add_filter( 'wp_default_scripts', array( $this, 'reregister_core_iapi_runtime' ), 20 );
+		}
+	}
+
+	/**
+	 * Re-registers the iAPI runtime registered by WordPress Core/Gutenberg, allowing WooCommerce to register its own version of the iAPI runtime.
+	 */
+	public function reregister_core_iapi_runtime() {
+		wp_deregister_script_module( '@wordpress/interactivity' );
+		wp_deregister_script_module( '@wordpress/interactivity-router' );
+
+		$interactivity_api_asset_data = $this->api->get_asset_data(
+			$this->api->get_block_asset_build_path( 'interactivity-api-assets', 'php' )
+		);
+
+		foreach ( $interactivity_api_asset_data as $handle => $data ) {
+			$handle_without_js = str_replace( '.js', '', $handle );
+			wp_register_script_module( $handle_without_js, plugins_url( $this->api->get_block_asset_build_path( $handle_without_js ), dirname( __DIR__ ) ), $data['dependencies'], $data['version'] );
+		}
```

Comment:
Can we be more defensive here? By only deregistering after ensuring the `$interactivity_api_asset_data` is valid.

Reply:
`get_asset_data` already validates the file exists, if it doesn't it returns an empty array and there is no-op.

So to be a bit more defensive here, I'll rearrange to check if `interactivity_api_asset_data` is more than an empty array, and contains entries for the 2 handles we de-register here and *then* deregister. Thanks its a good suggestion.

Reply:
Resolved here: https://github.com/woocommerce/woocommerce/pull/57611/commits/ae3731fcacd5418a82d19ae10bd2fcbb904f4c86

Note I could have just deregistered every `handle_without_js` but just to be extra careful I'm explicitly only deregistering handles that match the core iapi runtimes.